pkg/java/class.go

```diff
@@ -68,37 +69,77 @@ func HashClassInstructions(classBytes []byte) (string, error) {
 				for i < len(code) {
 					opcode := code[i]
 					h.Write([]byte{opcode})
-
-					// Look in the opcode tables to see how many operands
-					// this opcode takes, and advance to the end which must
-					// be another opcode or the end of the bytecode
-					if opcodes.NoOperandOpcodeLookupTable[opcode] {
-						i++
-					} else if opcodes.SingleOperandOpcodeLookupTable[opcode] {
-						i += 2
-					} else if opcodes.DoubleOperandOpcodeLookupTable[opcode] {
-						i += 3
-					} else if opcodes.QuadOperandOpcodeLookupTable[opcode] {
-						i += 5
-					} else {
-						for _, tripleOpcode := range opcodes.TripleOperandOpcodes {
-							if opcode == tripleOpcode {
-								i += 4
-								continue
-							}
-						}
-						// These opcodes take a variable amount of data and are not used
-						// in log4j. We're ignoring them for now as a result.
-						for _, otherOpcode := range opcodes.OtherOpcodes {
-							if opcode == otherOpcode {
-								return "", errors.New("unsupported opcode type")
-							}
-						}
-						return "", errors.New("unrecognised opcode")
+					operands, err := opcodeOperands(opcode, opcodes)
+					if err != nil {
+						return "", err
 					}
+					i += operands + 1
 				}
 			}
 		}
 	}
 	return fmt.Sprintf("%x-v0", h.Sum(nil)), nil
 }
+
+func ExtractBytecode(classBytes []byte) ([][]byte, error) {
+	classFile, err := classfile.Parse(classBytes)
+
+	if err != nil {
+		return nil, err
+	}
+
+	bytecode := make([][]byte, 0)
+	opcodes := OpcodeLookupTables()
+
+	for _, method := range classFile.Methods {
+		for _, attribute := range method.AttributeTable {
+			switch t := attribute.(type) {
+			default:
+				// ignore
+			case classfile.CodeAttribute:
+				code, i, extracted := t.Code, 0, bytes.Buffer{}
+				for i < len(code) {
+					opcode := code[i]
+					extracted.WriteByte(opcode)
+
+					operands, err := opcodeOperands(opcode, opcodes)
+					if err != nil {
+						return nil, err
+					}
+					i += operands + 1
+				}
+				bytecode = append(bytecode, extracted.Bytes())
+			}
+		}
+	}
+	return bytecode, nil
+}
+
+func opcodeOperands(opcode byte, opcodes Opcodes) (int, error) {
```

Comment:
nit: shall we make this a method with an `Opcodes` receiver, also let's locate home it in the `opcodes.go` file while we're extracting it?